vm-setup/roles/v1aX_integration_test/tasks/node_reuse.yml

```diff
@@ -335,22 +429,34 @@
         kubectl scale kubeadmcontrolplane "{{ CLUSTER_NAME }}" -n "{{ NAMESPACE }}" --replicas=3
     environment:
       KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+        #- name: Scale controlplane up back to 3.
+        #  k8s_scale:
+        #    api_version: v1alpha3
+        #    kind: KubeadmControlPlane
+        #    namespace: "{{ NAMESPACE }}"
+        #    name: "{{ CLUSTER_NAME }}"
+        #    kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+        #    replicas: 3
 
   - name: Wait until all "{{ NUMBER_OF_ALL_BMH }}" BMHs' are provisioned.
-    shell: kubectl get bmh -n "{{ NAMESPACE }}" | grep -w provisioned | wc -l
-    environment:
-      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    k8s_info:
+      api_version: metal3.io/v1alpha1
+      kind: BareMetalHost
+      namespace: "{{ NAMESPACE }}"
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: provisioned_host
     retries: 200
     delay: 20
-    register: provisioned_host
-    until: provisioned_host.stdout == "4"
+    until: provisioned_host.resources | filter_provisioning("provisioned") | length == 4
 
   - name: Wait until all "{{ NUMBER_OF_ALL_BMH }}" machines become running.
-    shell: |
-        kubectl get machines -n "{{ NAMESPACE }}" -o json | jq -r '.items[] | select (.status.phase == "Running") | .status.phase' | grep -c "Running"
-    environment:
-      KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    k8s_info:
+      api_version: cluster.x-k8s.io/v1alpha3
+      kind: machines
+      namespace: "{{ NAMESPACE }}"
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
     register: running_machines
     retries: 200
     delay: 20
-    until: running_machines.stdout|int == 4
\ No newline at end of file
+    until: running_machines.resources | filter_phase("running") | length == 4
```

Comment:
```suggestion
    until: running_machines.resources | filter_phase("running") | length == (NUM_NODES | int)
```

Reply:
I think here it is better to use `NUMBER_OF_ALL_BMH`, isn't it ?